packages/atomic/src/components/search/atomic-generated-answer/atomic-generated-answer.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  SearchStatus,
+  SearchStatusState,
+  buildSearchStatus,
+  buildGeneratedAnswer,
+  GeneratedAnswer,
+  GeneratedAnswerState,
+} from '@coveo/headless';
+import {Component, h, State} from '@stencil/core';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {Button} from '../../common/button';
+import {Bindings} from '../atomic-search-interface/atomic-search-interface';
+import {FeedbackButton} from './feedback-button';
+import {SourceCitations} from './source-citations';
+import {TypingLoader} from './typing-loader';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-generated-answer',
+  styleUrl: 'atomic-generated-answer.pcss',
+  shadow: true,
+})
+export class AtomicGeneratedAnswer implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  public generatedAnswer!: GeneratedAnswer;
+  public searchStatus!: SearchStatus;
+
+  @BindStateToController('generatedAnswer')
+  @State()
+  private generatedAnswerState!: GeneratedAnswerState;
+
+  @BindStateToController('searchStatus')
+  @State()
+  private searchStatusState!: SearchStatusState;
+
+  @State()
+  public error!: Error;
+
+  @State()
+  hidden = true;
+
+  public initialize() {
+    this.generatedAnswer = buildGeneratedAnswer(this.bindings.engine);
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+  }
+
+  private get hasRetryableError() {
+    return (
+      !this.searchStatusState.hasError &&
+      this.generatedAnswerState.error?.isRetryable
+    );
+  }
+
+  private get hideComponent() {
+    const {isLoading, answer, citations} = this.generatedAnswerState;
+    return (
+      !(isLoading || answer !== undefined || citations.length) &&
+      !this.hasRetryableError
+    );
+  }
+
+  private renderContent() {
+    return (
+      <div part="generated-content">
+        <div class="flex items-center">
+          <div part="header-label" class="text-bg-blue">
```

Comment:
For accessibility, I recommend copying the DOM structure of `atomic-smart-snippet`, including the heading.